src/node_credentials.cc

```diff
@@ -33,11 +35,32 @@ bool linux_at_secure = false;
 
 namespace credentials {
 
-// Look up environment variable unless running as setuid root.
+#if !defined(__CloudABI__) && !defined(_WIN32)
+// Returns true if the current process has effective capabilities and the
+// passed-in capability is in that set.
+bool HasCapability(int capability) {
+  DCHECK(cap_valid(capability));
+  struct __user_cap_header_struct cap_header_data = {
+    _LINUX_CAPABILITY_VERSION_3, getpid()
+  };
+  struct __user_cap_data_struct cap_data;
+
+  if (syscall(SYS_capget, &cap_header_data, &cap_data) == -1) {
```

Comment:
The Linux documentation recommends to use `cap_get_proc`. Is it not possible here?

See:
https://man7.org/linux/man-pages/man2/capget.2.html
https://man7.org/linux/man-pages/man3/cap_get_proc.3.html

Reply:
I used `capget` in my original commit but ran into an error where the build machine needs to have `libcap-devel` installed (sudo dnf install -y libcap-devel). My latest commit, which I've now reverted, was just an attempt to see if I could get  away without having this installed. It is only the header that is required and we don't need to link against libcap when using `capget`. Just by looking at the `cap_get_proc` it seems this would require us to link against `libcap` which I thought might be something we want to avoid. But I'm definitely open to suggestions here as to which path to take. Thanks